hugegraph-core/src/main/java/com/baidu/hugegraph/util/Reflection.java

```diff
@@ -0,0 +1,113 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.baidu.hugegraph.util;
+
+import com.baidu.hugegraph.HugeException;
+import com.baidu.hugegraph.exception.NotSupportException;
+import org.slf4j.Logger;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+public class Reflection {
+
+    private static final Logger LOG = Log.logger(Reflection.class);
+
+    private static final Class reflectionClazz;
+    private static final Method registerFieldsToFilterMethod;
+    private static final Method registerMethodsToFilterMethod;
+
+    public static final String JDK_INTERNAL_REFLECT_REFLECTION =
+                               "jdk.internal.reflect.Reflection";
+    public static final String SUN_REFLECT_REFLECTION =
+                               "sun.reflect.Reflection";
+
+    static {
+        Method registerFieldsToFilterMethodTemp = null;
+        Method registerMethodsToFilterMethodTemp = null;
+        Class reflectionClazzTemp = null;
+        try {
+            reflectionClazzTemp = Class.forName(
+                                        JDK_INTERNAL_REFLECT_REFLECTION);
+        } catch (ClassNotFoundException e) {
+            try {
+                reflectionClazzTemp = Class.forName(SUN_REFLECT_REFLECTION);
+            } catch (ClassNotFoundException ex) {
+                LOG.error("Can't find Reflection class", ex);
+            }
+        }
+
+        reflectionClazz = reflectionClazzTemp;
+
+        if (reflectionClazz != null) {
+            try {
+                registerFieldsToFilterMethodTemp =
+                reflectionClazz.getMethod("registerFieldsToFilter",
+                                          Class.class, String[].class);
+            } catch (Throwable e) {
+                LOG.error("Can't find registerFieldsToFilter method", e);
+            }
+
+            try {
+                registerMethodsToFilterMethodTemp =
+                reflectionClazz.getMethod("registerMethodsToFilter",
+                                          Class.class, String[].class);
+            } catch (NoSuchMethodException e) {
+                LOG.error("Can't find registerMethodsToFilter method", e);
+            }
+        }
+        registerFieldsToFilterMethod = registerFieldsToFilterMethodTemp;
+        registerMethodsToFilterMethod = registerMethodsToFilterMethodTemp;
+    }
+
+    public static void registerFieldsToFilter(Class<?> containingClass,
+                                              String ... fieldNames) {
+        if (registerFieldsToFilterMethod == null) {
+            throw new NotSupportException(
+                      "No support this method 'registerFieldsToFilter'");
```

Comment:
seem NotSupportException don't need to pass full error message into

Reply:
do you mean change to "throw new NotSupportException()"?

Reply:
NotSupportException has prefix message, we can change it to:
 `throw new NotSupportException("Reflection.registerFieldsToFilter()")`